src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -351,6 +358,16 @@ protected virtual void ClearProjectData(ProjectId projectId)
         protected internal virtual void ClearDocumentData(DocumentId documentId)
             => this.ClearOpenDocument(documentId);
 
+        /// <summary>
+        /// Allows workspaces to register themselves to be considered when LSP requests come in. Any workspace
+        /// registered will be probed for a matching document/solution which can be given to the request handler
+        /// to operate on.
+        /// </summary>
+        protected void RegisterForLsp()
```

Comment:
If we're putting this API in the Workspace layer, spell out "Lsp", since otherwise this is a TLA that nobody will understand.